controllers/observabilityendpoint/ocp_monitoring_config_test.go

```diff
@@ -0,0 +1,185 @@
+		t.Fatalf("empty AlertmanagerConfigs in ClusterMonitoringConfiguration.PrometheusK8sConfig: %v", foundClusterMonitoringConfiguration)
+	}
+
+	containsOCMAlertmanagerConfig := false
+	for _, v := range foundClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs {
+		if v.TLSConfig != (cmomanifests.TLSConfig{}) &&
+			v.TLSConfig.CA != nil &&
+			v.TLSConfig.CA.LocalObjectReference != (corev1.LocalObjectReference{}) &&
+			v.TLSConfig.CA.LocalObjectReference.Name == hubAmRouterCASecretName &&
+			v.BearerToken != nil &&
+			v.BearerToken.LocalObjectReference != (corev1.LocalObjectReference{}) &&
+			v.BearerToken.LocalObjectReference.Name == hubAmAccessorSecretName {
+			containsOCMAlertmanagerConfig = true
+			foundHubAmAccessorSecret := &corev1.Secret{}
+			err = c.Get(ctx, types.NamespacedName{Name: v.BearerToken.LocalObjectReference.Name,
+				Namespace: promNamespace}, foundHubAmAccessorSecret)
+			if err != nil {
+				t.Fatalf("failed to check the observability-alertmanager-accessor secret %s: %v", clusterMonitoringConfigName, err)
+			}
+			foundAmAccessorToken, ok := foundHubAmAccessorSecret.Data[hubAmAccessorSecretKey]
+			if !ok {
+				t.Fatalf("no key %s found in the observability-alertmanager-accessor secret", hubAmAccessorSecretKey)
+			}
+			if string(foundAmAccessorToken) != testBearerToken {
+				t.Fatalf("incorrect token found in the observability-alertmanager-accessor secret, got token: %s, expected value %s", foundAmAccessorToken, testBearerToken)
+			}
+		}
+	}
+
+	if containsOCMAlertmanagerConfig == false {
+		t.Fatalf("no AlertmanagerConfig for OCM in ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs: %v", foundClusterMonitoringConfiguration)
+	}
+
+	err = revertClusterMonitoringConfig(ctx, c)
+	if err != nil {
+		t.Fatalf("Failed to revert cluster-monitoring-config configmap: (%v)", err)
+	}
+
+	err = c.Get(ctx, types.NamespacedName{Name: clusterMonitoringConfigName,
+		Namespace: promNamespace}, foundCusterMonitoringConfigMap)
+	if err == nil || !errors.IsNotFound(err) {
+		t.Fatalf("the configmap %s should be deleted", clusterMonitoringConfigName)
+	}
+
+	foundHubAmAccessorSecret := &corev1.Secret{}
+	err = c.Get(ctx, types.NamespacedName{Name: hubAmAccessorSecretName,
+		Namespace: promNamespace}, foundHubAmAccessorSecret)
+	if err == nil || !errors.IsNotFound(err) {
+		t.Fatalf("the secret %s should be deleted", hubAmAccessorSecretName)
+	}
+
+	foundHubAmRouterCASecret := &corev1.Secret{}
+	err = c.Get(ctx, types.NamespacedName{Name: hubAmRouterCASecretName,
+		Namespace: promNamespace}, foundHubAmRouterCASecret)
+	if err == nil || !errors.IsNotFound(err) {
+		t.Fatalf("the secret %s should be deleted", hubAmRouterCASecretName)
+	}
+
+	err = revertClusterMonitoringConfig(ctx, c)
+	if err != nil {
+		t.Fatalf("Run into error when try to revert cluster-monitoring-config configmap twice: (%v)", err)
+	}
```

Comment:
You may add more test cases to cover:
1. has `cluster-monitoring-config` w/o prometheusK8s
2. has `cluster-monitoring-config` w/ prometheusK8s
3. has `cluster-monitoring-config` w/o additionalAlertManagerConfigs
4. 3. has `cluster-monitoring-config` w additionalAlertManagerConfigs

Reply:
thanks, added more cases for `cluster-monitoring-config` reconcile process.